frontend/src/pages/functional_components/RecurringRunDetailsV2FC.tsx

```diff
@@ -33,75 +32,120 @@ import { triggerDisplayString } from 'src/lib/TriggerUtils';
-          message:
-            `Error: failed to retrieve recurring run: ${recurringRunId}.` +
-            (errorMessage ? ' Click Details for more information.' : ''),
-          mode: 'error',
-        });
-        return;
-      }
+      const recurringRun = await Apis.recurringRunServiceApi.getRecurringRun(recurringRunId);
+      setRecurringRunName(recurringRun.display_name);
+      setRecurringRunStatus(recurringRun.status);
+      setRecurringRunIdFromApi(recurringRun.recurring_run_id);
+
+      return recurringRun;
     },
     { enabled: !!recurringRunId, staleTime: 0, cacheTime: 0 },
   );
 
   const experimentId = recurringRun?.experiment_id!;
-  const { data: experiment } = useQuery<V2beta1Experiment | undefined, Error>(
+  const { error: getExperimentError } = useQuery<void, Error>(
     ['experiment'],
     async () => {
-      try {
-        return await Apis.experimentServiceApiV2.getExperiment(experimentId);
-      } catch (err) {
-        const errorMessage = await errorToMessage(err);
-        updateBanner({
-          additionalInfo: errorMessage ? errorMessage : undefined,
-          message:
-            `Error: failed to retrieve this recurring run's experiment.` +
-            (errorMessage ? ' Click Details for more information.' : ''),
-          mode: 'warning',
-        });
-        return;
-      }
+      const experiment = await Apis.experimentServiceApiV2.getExperiment(experimentId);
+      setExperimentName(experiment.display_name);
+      setExperimentIdFromApi(experiment.experiment_id);
+      // The logic related to experiment is done in useQuery, so no need to return.
     },
     { enabled: !!experimentId, staleTime: 0 },
   );
 
   useEffect(() => {
-    setToolbarState(getInitialToolbarState());
+    const toolbarState = getInitialToolbarState();
+
+    toolbarState.actions[ButtonKeys.ENABLE_RECURRING_RUN].disabled =
+      recurringRunStatus === V2beta1RecurringRunStatus.ENABLED;
+    toolbarState.actions[ButtonKeys.DISABLE_RECURRING_RUN].disabled =
+      recurringRunStatus !== V2beta1RecurringRunStatus.ENABLED;
+    toolbarState.pageTitle = recurringRunName || recurringRunIdFromApi || 'Unknown recurring run';
+    toolbarState.breadcrumbs = getBreadcrumbs(experimentIdFromApi, experimentName);
+    updateToolbar(toolbarState);
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [recurringRun]);
+  }, [
+    recurringRunIdFromApi,
+    recurringRunName,
+    recurringRunStatus,
+    experimentName,
+    experimentIdFromApi,
+  ]);
+
+  useEffect(() => {
+    if (getRecurringRunError) {
+      (async () => {
+        const errorMessage = await errorToMessage(getRecurringRunError);
+        setGetRecurringRunErrMsg(errorMessage);
+      })();
+    }
+
+    if (getExperimentError) {
+      (async () => {
+        const errorMessage = await errorToMessage(getExperimentError);
+        setGetExperimentErrMsg(errorMessage);
+      })();
+    }
+  }, [getRecurringRunError, getExperimentError]);
 
```

Comment:
When `effect` logics are independent between `getRecurringRunError` and `getExperimentError`, let's have two separate `useEffect` to maintain atomicity. 

```
  useEffect(() => {
    if (getRecurringRunError) {
      (async () => {
        const errorMessage = await errorToMessage(getRecurringRunError);
        setGetRecurringRunErrMsg(errorMessage);
      })();
    }
  }, [getRecurringRunError]);

  useEffect(() => {
    if (getExperimentError) {
      (async () => {
        const errorMessage = await errorToMessage(getExperimentError);
        setGetExperimentErrMsg(errorMessage);
      })();
    }
  }, [getExperimentError]);
```

Reply:
Added documentation for the original implementation.